lib/build/geo_feature_builder.dart

```diff
@@ -0,0 +1,201 @@
+import 'dart:core';
+import 'dart:io';
+import 'dart:convert';
+import 'package:build/build.dart';
+import 'package:code_builder/code_builder.dart';
+import 'package:dart_style/dart_style.dart';
+import 'package:recase/recase.dart';
+
+class GeoFeatureBuilder implements Builder{
+
+  final BuilderOptions options;
+
+  GeoFeatureBuilder(this.options);
+  
+  @override
+  Map<String, List<String>> get buildExtensions => {
+    options.config['source_file'] : [ options.config['output_file'] ],
+  };
+
+  @override
+  Future<void> build(BuildStep buildStep) async {
+    final AssetId inputId = buildStep.inputId;
+    try {
+        final geoJSONData = await _readFile(File(inputId.path));
+        final propertiesToExtract = options.config['extract_properties'].map<String,String>(
+            (key, value) => MapEntry<String, String>(key.toString(), value.toString())
+        );
```

Comment:
Either
`final Map<String, String> propertiesToExtract = options.config['extract_properties'];`
Or
`final propertiesToExtract = options.config['extract_properties'].cast<String, String>();`
should do the trick.